### PR TITLE
fix: block odp methods on datafile

### PR DESCRIPTION
**Pull request by andrewleap-optimizely**
Summary
-------

- Added blocking call to `Optimizely._identify`, `Optimizely._fetch_qualified_segments` and `Optimizely.send_odp_event` to ensure datafile has been downloaded before attempting to make odp requests.

Test plan
---------
all existing tests pass

### CODE REVIEW

File: optimizely/optimizely.py [hunk 1]
```diff
@@ -1357,13 +1357,27 @@ def _identify_user(self, user_id: str) -> None:
             self.logger.error(enums.Errors.INVALID_OPTIMIZELY.format('identify_user'))
             return
 
+        if not self.odp_manager.enabled:
+            self.logger.debug('ODP identify event is not dispatched (ODP disabled).')
+            return
```
Comment (jaeopt) [inline optimizely/optimizely.py:1362]: I believe this is checked in odp_manager. Do we need to check here again?

**Review by jaeopt — APPROVED** [review]
LGTM - a clarification

**Review by Mat001 — APPROVED** [review]
Looks good. Tested fetch without calling opti config.

File: optimizely/optimizely.py (hunk 1 above)
Comment (andrewleap-optimizely) [reply optimizely/optimizely.py:1362]: This allows `create_user_context` to remain non-blocking if odp is disabled. Seems like a worthwhile optimization?

Comment (jaeopt) [reply optimizely/optimizely.py:1362]: @andrewleap-optimizely I see. Probably not in server sdks. They'll be blocked anyway in following decide call even if avoid it here :)

**Pull request merged**